### PR TITLE
Initial version of debezium-server plugin to export data to files

### PR DESCRIPTION
**Pull request by makalaaneesh**
@amit-yb 
1. Created a branch called voyager-main that is up to date with main debezium repo master as of now. 
2. Included the tests from the debezium-server-http module as of now for reference for when we write tests.

### CODE REVIEW

File: debezium-server/debezium-server-ybexporter/src/main/java/io/debezium/server/ybexporter/TableSnapshotWriterCSV.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    private CSVPrinter csvPrinter;
+
+    public TableSnapshotWriterCSV(String datadirStr, Table tbl) {
+        dataDir = datadirStr;
+        t = tbl;
+
+        var fileName = getFullFileNameForTable();
+        try {
+            var f = new FileWriter(fileName);
+            csvPrinter = new CSVPrinter(f, CSVFormat.POSTGRESQL_CSV);
```
Comment (amit-yb) [inline debezium-server/debezium-server-ybexporter/src/main/java/io/debezium/server/ybexporter/TableSnapshotWriterCSV.java:31]: Can we use `CSVFormat.POSTGRESQL_TEXT`?

**Review by amit-yb — APPROVED** [review]
LGTM.

As a general guideline, in the critical path:
- Be watchful of every allocation
- Avoid logging.
- reuse objects from previous iteration, as much as possible.

File: debezium-server/debezium-server-dist/cpfile.sh [hunk 2]
```diff
@@ -0,0 +1 @@
+cp ~/Documents/debezium/debezium-server/debezium-server-ybexporter/target/debezium-server-ybexporter-2.1.0-SNAPSHOT.jar ~/Documents/debezium/debezium-server/debezium-server-dist/target/debezium-server/lib/
```
[the remark this answers is not in the document]
Comment (makalaaneesh) [reply debezium-server/debezium-server-dist/cpfile.sh:1]: Didn't realise this got committed. Will remove .

File: debezium-server/debezium-server-ybexporter/src/main/java/io/debezium/server/ybexporter/Record.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    public String op;
+    public HashMap<String, Object> keyFields = new HashMap<>();
+    public LinkedHashMap<String, Object> valueFields = new LinkedHashMap<>();
+
+    public String getTableIdentifier() {
+        return t.toString();
+    }
+
+    public ArrayList<Object> getValueFieldValues() {
+        return new ArrayList<>(valueFields.values());
```
[the remark this answers is not in the document]
Comment (makalaaneesh) [reply debezium-server/debezium-server-ybexporter/src/main/java/io/debezium/server/ybexporter/Record.java:24]: Yep, agreed! I've already made this modification in the optimizations branch 👍

File: debezium-server/debezium-server-ybexporter/src/main/java/io/debezium/server/ybexporter/TableSnapshotWriterCSV.java (hunk 1 above)
Comment (makalaaneesh) [reply debezium-server/debezium-server-ybexporter/src/main/java/io/debezium/server/ybexporter/TableSnapshotWriterCSV.java:31]: Will explore/test TEXT format in my work on optimizations. 
Is the motivation strictly performance based? Or any other reason why we'd prefer TEXT over CSV?

Comment (makalaaneesh) [reply debezium-server/debezium-server-ybexporter/src/main/java/io/debezium/server/ybexporter/TableSnapshotWriterCSV.java:31]: Tested out TEXT format. For some reason, the format isn't working as expected - using COPY command with format TEXT wasn't working. Had to eventually use CSV with specific options to get it to work. (`COPY test_basic FROM '/tmp/export-dir/test_basic_data.sql'  CSV DELIMITER E'\t' ESCAPE '\' NULL '\N';`)

**Pull request merged**